### DISCUSSION TITLE
How URI resolution will work if `$id` keyword is an empty string? Or if multiple schemas have the same canonical URI?

### DISCUSSION THREAD

**Discussion by flaksp** [json-schema-org/community#259] [Q&A] [answered]
Seems like `$id` may be an empty string:

```
"uriReferenceString": {
  "type": "string",
  "format": "uri-reference"
}
```

So, this document contains 2 schemas:

```json
{
  "$id": "https://example.com/Foo",
  "$defs": {
    "bar": {
      "$id": ""
    }
  }
}
```

Does `https://example.com/Foo` references the root schema or the `bar` schema? What behaviour should be if multiple schemas have the same canonical URI?

**Comment by handrews** [comment, marked as answer]
`""` is a relative URI-reference with a path component consisting of an empty string.  Setting `"$id": ""` like this is technically an error in accordance with [§9.1.2](https://www.ietf.org/archive/id/draft-bhutton-json-schema-01.html#section-9.1.2):

> A schema MAY (and likely will) have multiple URIs, but there is no way for a URI to identify more than one schema. When multiple schemas try to identify as the same URI, validators SHOULD raise an error condition.

because resolving `""` against the base URI of `""https://example.com/Foo"` just produces `"https://example.com/Foo"` again, meaning that both of these `"$id"`s attempt to assign the same URI to two different schemas.

**Reply by handrews** [reply]
This was recently somewhat clarified by issue json-schema-org/json-schema-spec#1271 and PR json-schema-org/json-schema-spec#1272, BTW.
